src/bp/ui-admin/src/reducers/server.ts

```diff
@@ -0,0 +1,59 @@
+import api from '../api'
+
+export const FETCH_LANGUAGES_RECEIVED = 'server/FETCH_LANGUAGES_RECEIVED'
+export const FETCH_WORKSPACES_RECEIVED = 'server/FETCH_WORKSPACES_RECEIVED'
+export const FETCH_SERVER_CONFIG_RECEIVED = 'server/FETCH_FEATURE_RECEIVED'
+
+const initialState = {
+  languages: null,
+  workspaces: null,
+  serverConfig: null,
+  serverConfigLoaded: false
+}
+
+export default (state = initialState, action) => {
+  switch (action.type) {
+    case FETCH_LANGUAGES_RECEIVED:
+      return {
+        ...state,
+        languages: action.languages
+      }
+
+    case FETCH_WORKSPACES_RECEIVED:
+      return {
+        ...state,
+        workspaces: action.workspaces
+      }
+
+    case FETCH_SERVER_CONFIG_RECEIVED:
+      return {
+        ...state,
+        serverConfig: action.serverConfig,
+        serverConfigLoaded: true
+      }
+
+    default:
+      return state
+  }
+}
+
+export const fetchLanguages = () => {
```

Comment:
you moved this here but did you make the changes in the server language page  ?

Reply:
I converted the server.js to server.ts, and added fetchWorkspaces and fetchServerConfig. Language was already there

Reply:
I converted the server.js to server.ts, and added fetchWorkspaces and fetchServerConfig. Language was already there